File: languages.py
import requests
import json
import os

data_str="""{
    "About": {
        "zh": "关于"
    },
    "About the App": {
        "zh": "退出APP"
    },
    "Clear Context": {
        "zh": "清除所有上下文"
    }
}"""

langs_data=json.loads(data_str)

def UText(k):
    l = os.getenv("ULANG") if os.getenv("ULANG") else "zh"
    if k in langs_data:
        if l in langs_data[k]:
            return langs_data[k][l] if langs_data[k][l] else k
        else:
            print(f"not supported lang: {l}")
            request_online(k,l)
            return k
    else:
        request_online(k,l)
        return k

def request_online(k,l):
    api_addr = "http://127.0.0.1:1850"
    project_id = "123"
    url = f"{api_addr}/{project_id}/{k}/{l}"  # Replace with the actual URL of your API endpoint

    print(url)
    response = requests.get(url)

    if response.status_code == 200:
        result = response.json()["result"]
        return result
    else:
        print(f"Failed to get data. Status code: {response.status_code}")

File: templates/languages.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
)

var langs_json = `{languages_json}`
var langs_data map[string]map[string]string

func UText(k string) string {
	if len(langs_data) == 0 {
		json.Unmarshal([]byte(langs_json), &langs_data)
	}
	l := os.Getenv("ULANG")
	if l == "" {
		l = "default"
	}
	if val, ok := langs_data[k]; ok {
		if v, ok := val[l]; ok {
			return v
		} else {
			fmt.Printf("Not supported lang: %v\n", l)
			go requestOnline(k, l)
		}
	} else {
		go requestOnline(k, l)
	}
	return k
}

func requestOnline(k, l string) {
	url := fmt.Sprintf("{api_addr}/{project_id}/key?k=%v&l=%v", k, l)

	resp, err := http.Get(url)
	if err != nil {
		fmt.Printf("Failed to get data: %v\n", err)
		return
	}
	defer resp.Body.Close()

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Printf("Failed to read data: %v\n", err)
		return
	}

	var result map[string]string
	json.Unmarshal(body, &result)

	return
}
